pkg/kubelet/dockertools/manager.go

```diff
@@ -276,6 +276,38 @@ type containerStatusResult struct {
 	err    error
 }
 
+// determineInfraContainerIP inspects the given container and determines the IP address of that
+// container.  The container is assumed to be the pod infra container; it is the responsibility
+// of the caller to ensure that the container ID passed is that of the infra container.
+func (dm *DockerManager) determineInfraContainerIP(dockerID string, pod *api.Pod) (string, error) {
+	inspectResult, err := dm.client.InspectContainer(dockerID)
+	if err != nil {
+		return "", err
+	}
+	if inspectResult == nil {
+		return "", nil // TODO: should be an error, probably
```

Comment:
This if clause is bothering me; `inspectContainer` does it but it seems like it would represent a bug in the docker API to get an `(empty container, nil)` result from `InspectContainer`.  It would be great if there were a comment on the original code with a reference to an issue either in kube or docker, but oh well.  I'm strongly tempted to take this clause out...